src/command_line_parser.cc

```diff
@@ -1688,13 +1716,14 @@ CLParser::ParseCommandLine(int argc, char** argv)
       }
     }
     catch (const std::invalid_argument& ia) {
-      if (opt >= 'A') {  // short options
+      if (opt < long_option_idx_base) {  // short options
         Usage(
             "Failed to parse -" + std::string{(char)opt} +
             ". Invalid value provided: " + std::string{optarg});
       } else {
         Usage(
-            "Failed to parse --" + std::string{long_options[opt].name} +
+            "Failed to parse --" +
+            std::string{long_options[opt - long_option_idx_base].name} +
```

Comment:
In order to get the name of long options now, we need to subtract away the base index to get the actual index of the long option we want in the `getopt.h` data structure for holding long options.